content/building-blocks/automate-and-execute-your-work/reproducible-work/dockerhub.md

```diff
@@ -0,0 +1,200 @@
+    volumes:
+      - type: bind
+        source: ~/your-env
+        target: /your-env
+```
+
+{{% /codeblock %}}
+
+For the sake of this building block, if you introduce the appropriate repository name, the tag of the image you are interested in, and the port where you want jupyter to be executed (e.g. 4444) this docker-compose will be ready to allow you to work with your imported environment. Again, feel free to copy it and use it as your docker-compose template! It has been designed to work properly in conjunction with the dockerfile above by following the steps within this building block. For that you should save it (for example, in the same directory where you placed your dockerfile previously) as "docker-compose.yml", open the command line there and run:
+
+`docker compose up`
+
+Now you just have to access jupyter by clicking on the link generated in your command line, or by copying it into your browser's search bar.
+
+<p align = "center">
+<img src = "../img/command_line_jupyter.png" width="750">
+<figcaption> Click on the generated link or paste it into your browser to access jupyter. </figcaption>
+</p>
+
+If the execution was successful, once you access jupyter you will notice that it has been opened in a directory called "your-env". This directory exists in both your local computer and the image's container and is the connection between the two. This is the main function of the [docker volume](https://docs.docker.com/storage/volumes/) that was set in the docker-compose and it implies that any file you place on this directory on your pc will be made available also for you to work with it in your environment's container. Conversely, you can retrieve from the container to your local filesystem any file generated which is stored here. Ideally, here is where you should place the files of your project so you can work with them more flexibly.
+
+You can also use any name for this directory where the volume has been mounted (first in the dockerfile and then in the docker-compose) to make it more recognizable to you.
+
+<p align = "center">
+<img src = "../img/dockerfile-your-env.png" width="400">
+</p>
+
+<p align = "center">
+<img src = "../img/docker-compose-your-env.png" width="500">
+<figcaption> Change the name of the directory where the volume is mounted to whichever you prefer. Just make sure that it is the same across both the dockerfile and docker-compose files. </figcaption>
+</p>
+
+### Creating new versions of your images with "docker commit"
+
+Imagine that during your project, have started to use a new python package not included in the original image that you already made available to other users on Docker Hub. In that case, if you have this package already installed in your image's container, you could simply use the [docker commit](https://docs.docker.com/engine/reference/commandline/commit/) command to create a new version of your image including these changes and then push it. For that you just have to run:
```

Comment:
Didn't know this @DiSanchz , neat addition :D

Reply:
Thank you! I thought it was an interesting addition to the BB